metadata_layer.py

```diff
@@ -148,6 +149,15 @@ def append_only(self, **args):
             **args
         )
 
+    def mirror_type_for(self, link):
+        """Returns the type of mirror that should be used for the link.
+        """
+        if link.rel in [Hyperlink.IMAGE, Hyperlink.THUMBNAIL_IMAGE]:
+            return ExternalIntegrationLink.COVERS
+        elif link.rel in Hyperlink.CIRCULATION_ALLOWED:
+            return ExternalIntegrationLink.BOOKS
+        return None
+
```

Comment:
This doesn't use `self` at all. As is, this would be a better fit in `LinkData`. It would make sense as part of `ReplacementPolicy` if it looked in `self.mirrors` and returned the actual mirror object to use.

Normally I don't think this is a big deal but `metadata_layer.py`, like the model, is a semi-public API that's used by a lot of different code so I'm more concerned with the API design.